profilingdemo/source/java/src/org/kaaproject/kaa/demo/profiling/ProfilingDemo.java

```diff
@@ -0,0 +1,198 @@
+ */
+public class ProfilingDemo {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProfilingDemo.class);
+
+    public static final int KAA_CLIENT_NUMBER = 3;
+    /**
+     * In this dir all Kaa clients creates it's keys for starting
+     */
+    public static final String PROPERTIES_OUT_KEYS_DIR = "./res/out/kaaTempDir";
+
+    private static ArrayList<KaaClient> kaaClients = new ArrayList<>();
+    private static ArrayList<InnerData> profiles = new ArrayList<>();
+
+    public static void main(String[] args) throws IOException {
+        LOG.info("Profiling demo started");
+
+        for (int i = 0; i < KAA_CLIENT_NUMBER; i++) {
+            try {
+                startKaaClient(i);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException was caught - ", e);
+        }
+
+        // Stop the Kaa client and release all the resources which were in use.
+        try {
+            stopKaaClients();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        LOG.info("Profiling demo stopped");
+    }
+
+    public static void startKaaClient(final int index) throws IOException {
+
+        KaaClientProperties kaaClientProperties = new KaaClientProperties();
+        kaaClientProperties.setWorkingDirectory(PROPERTIES_OUT_KEYS_DIR + index);
+
+        // Create the Kaa desktop context for the application.
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext(kaaClientProperties);
+
+        // Create a Kaa client and add a listener which displays the Kaa client profiling
+        // as soon as the Kaa client is started.
+        final KaaClient kaaClient = Kaa.newClient(desktopKaaPlatformContext, new SimpleKaaClientStateListener() {
+            @Override
+            public void onStarted() {
+                super.onStarted();
+
+                displayConfiguration(index);
+            }
+        }, true);
+        kaaClients.add(kaaClient);
+
+        kaaClient.setProfileContainer(new ProfileContainer() {
+            @Override
+            public PagerClientProfile getProfile() {
+                InnerData data = init().get(index);
+
+                PagerClientProfile pagerClientProfile = new PagerClientProfile();
+                pagerClientProfile.setAudioSupport(data.isAudio());
+                pagerClientProfile.setVibroSupport(data.isVibro());
+                pagerClientProfile.setVideoSupport(data.isVideo());
+
+                return pagerClientProfile;
+            }
+        });
+
+        // Persist profiling in a local storage to avoid downloading it each time the Kaa client is started.
+        kaaClient.setConfigurationStorage(new SimpleConfigurationStorage(desktopKaaPlatformContext, "saved_config.cfg"));
```

Comment:
I faced funny bug: I had only one `saved_config.cfg` file in an application's work. dir. I assume that you need to specify full path to file, for example `"./clientWorkDir_" + index "/saved_config.cfg"`.